### PR TITLE
lua-openssl: Upgrade to v0.7.4

### PR DESCRIPTION
**Pull request by amnonpaz**
Maintainer: Amnon Paz @amnonpaz 
Compile tested: x86, mips
Tested: x86 emulation

Description:
Update for lua-openssl library. Basically, patch the library Makefile so it compiles with OpenWrt

### CODE REVIEW

**Comment by amnonpaz** [comment]
@diizzyy Please review

File: lang/lua-openssl/Makefile [hunk 1]
```diff
@@ -8,16 +8,16 @@
...
 PKG_LICENSE:=MIT
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_MIRROR_HASH:=225e3fe09226ff72968a7f36a33e207d27332107456a754abbaa59f99a3038f3
+PKG_MIRROR_HASH:=c27cedee438de95877823b1ae0607556564d82c8692be40f43743ca9cc5a029a
 PKG_SOURCE_URL:=https://github.com/zhaozg/lua-openssl.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_VERSION:=b104bbe914d279276560f188854036075b99f724
+PKG_SOURCE_VERSION:=96effbaf477ca205f1787aaf1cce643bd208066b
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
```
Comment (diizzyy) [inline lang/lua-openssl/Makefile]: Perferably switch to codeload

**Comment by diizzyy** [comment]
I don't use lua so I'm probably not the correct person to ask ;-)
As far as I can tell it looks good however I think you should try to fix upstream Makefile instead of more or less rewriting it.

@karlp 
Perhaps you can have a look?

**Comment by karlp** [comment]
I can't read diffs of diffs and don't use lua-openssl sorry.  @amnonpaz is already thhe maintainer of this package so I'd say probably fine.  I would suggest that the commit message include some sort of overview of theh changelog, or a link to the changelog upstream though.

File: lang/lua-openssl/Makefile (hunk 1 above)
Comment (amnonpaz) [reply lang/lua-openssl/Makefile]: What is codeload? (tried google it, nothing coherent went up...)

**Comment by amnonpaz** [comment]
@diizzyy You've got a point there, but the reason for this rewriting is that the original Makefile supports IOs, mingw and other platforms. The tools it uses for configuring the right platform (and versions) didn't work from within the openWRT build process.

@karlp I will update the commit message as you suggested

File: lang/lua-openssl/Makefile (hunk 1 above)
Comment (neheb) [reply lang/lua-openssl/Makefile]: git grep the packages repo. There are plenty of examples.

**Comment by neheb** [comment]
ping @amnonpaz is this ready?

**Comment by neheb** [comment]
New version is out. Please update.

**Pull request merged**